frontend/src/components/AllGroups.tsx

```diff
@@ -114,30 +114,56 @@ function AllGroupsContent() {
       {groups && groups.length > 0 ? (
         <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
           {groups.map((group: GroupType) => (
-            <div key={group.id} className="card bg-neutral text-neutral-content w-full shadow-xl">
+            <div
+              key={group.id}
+              className="card bg-neutral text-neutral-content w-full shadow-xl cursor-pointer transition-transform duration-200 ease-in-out transform hover:scale-[1.01] hover:shadow-md hover:shadow-primary/800"
+              onClick={() => router.push(`/group/${group.id}`)}
+            >
               <div className="card-body items-center text-center">
                 <h2 className="card-title">{group.name}</h2>
                 <p>Owner: {group.owner.name} (ID: {group.owner.id})</p>
-                <div className="card-actions justify-end">
-                  <button 
-                    className="btn btn-primary"
-                    onClick={() => router.push(`/group/${group.id}`)}
-                  >
-                    View
-                  </button>
+                <div className="card-actions justify-center space-x-4 mt-4">
                   <button
-                    className="btn btn-backgroundOffset"
-                    onClick={() => handleEditClick(group.id)}
+                    className="btn btn-primary flex items-center justify-center"
+                    onClick={(e) => {
+                      e.stopPropagation();
+                      handleEditClick(group.id);
+                    }}
                   >
-                    Edit
+                    <svg
```

Comment:
Since these SVGs are being used in multiple places, keep them in common place and access from there. Either keep in assets folder or create a component for it